ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/JsonRpcHttpService.java

```diff
@@ -161,6 +169,29 @@ private void validateConfig(final JsonRpcConfiguration config) {
     return resultFuture;
   }
 
+  private Handler<RoutingContext> checkWhitelistHostHeader() {
+    return event -> {
+      Optional<String> hostHeader =
+          Optional.ofNullable(Iterables.get(Splitter.on(':').split(event.request().host()), 0));
```

Comment:
What happens if an invalid Host header is sent that contains more than one colon?

Reply:
We take the first item from the split and do the checks on it.  If `*` or `all` are specified it'll get through, otherwise it'll get matched against the specified whitelist and _likely_ fail.

I think it'll still work as intended.

Reply:
I would have though it if the host was: `some:host:8080` you should require `some:host` to be in the whitelist and reject if only `some` is in the whitelist.  That said, I'm not sure if the request would make it through vertx at all since it's a pretty seriously invalid http header.